### PR TITLE
Bir/da 589 FAQ UI

### PR DESCRIPTION
**Pull request by MuBirhan**

### CODE REVIEW

File: marty_flutter/lib/screens/faq/faq_widget.dart [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    return ListView.separated(
+      padding: EdgeInsets.only(left: 0),
+      itemCount: this.data.length,
+      itemBuilder: (BuildContext context, int index) {
+        return CustomExpansionTile(data: this.data[index]);
+      },
+      separatorBuilder: (BuildContext context, int index) => Divider(
+        indent: Dimens.largePadding,
+        endIndent: Dimens.largePadding,
+        height: Dimens.dividerHeight,
```
Comment (ancamihesan) [inline marty_flutter/lib/screens/faq/faq_widget.dart]: @MuBirhan please rename Dimens to AppDimens

File: marty_flutter/lib/themes/app_theme.dart [hunk 2]
```diff
@@ -22,6 +23,12 @@ class AppFontStyles {
       fontSize: 20,
       fontWeight: FontWeight.w500);
 
+  static const mediumBlackText = TextStyle(
```
Comment (ancamihesan) [inline marty_flutter/lib/themes/app_theme.dart]: @MuBirhan please set the base theme for the app. More details here https://flutter.dev/docs/cookbook/design/themes

File: marty_flutter/lib/models/faq_data.dart [hunk 3]
```diff
@@ -0,0 +1,7 @@
+class FAQData {
+  String title;
```
Comment (vladms) [inline marty_flutter/lib/models/faq_data.dart]: @MuBirhan These params can be final

File: marty_flutter/lib/screens/faq/faq_elements.dart [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+class _ExpansionState extends State<CustomExpansionTile> {
+  FAQData data;
+  Image buttonImage = Image.asset('assets/dropdown_arrow_normal.png');
+  AnimationController controller;
+
+  _ExpansionState(this.data);
+
+  _expandWidget() {
+    setState(() {
+      if (this.data.expanded) {
```
Comment (vladms) [inline marty_flutter/lib/screens/faq/faq_elements.dart]: @MuBirhan this.data.expanded = !this.data.expanded;

**Comment by vladms** [comment]
@MuBirhan Just checked the UI:

The dividers between list cells have a left/right padding

File: marty_flutter/lib/screens/faq/faq_elements.dart [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+  final FAQData data;
+
+  const CustomExpansionTile({Key key, this.data}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() => _ExpansionState(this.data);
+}
+
+class _ExpansionState extends State<CustomExpansionTile> {
+  FAQData data;
```
Comment (vladms) [inline marty_flutter/lib/screens/faq/faq_elements.dart]: @MuBirhan This can be private I think

File: marty_flutter/lib/main.dart [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
 import 'package:marty_flutter/screens/shifts/my_shifts.dart';
 import 'package:marty_flutter/themes/app_colors.dart';
-import 'package:marty_flutter/themes/dimens.dart';
-import 'package:marty_flutter/widgets/rounded_corners_widget.dart';
-
+import 'package:marty_flutter/themes/app_dimens.dart';
+import 'package:marty_flutter/themes/icons.dart';
+import 'package:marty_flutter/themes/values.dart';
+import 'package:marty_flutter/widgets/main_widget.dart';
 import 'models/ScreenType.dart';
```
Comment (vladms) [inline marty_flutter/lib/main.dart]: @MuBirhan  This looks like a bad import

File: marty_flutter/lib/screens/faq/faq.dart [hunk 7]
```diff
@@ -1,22 +1,48 @@
...
-  FAQ()
+  //TODO: Change the screen titles after localization is done
+  FAQScreen()
       : super(
             "FAQ",
-            Container(
-              height: 400.0,
-              child: FAQWidget(data: dataList),
+            ListView.separated(
+              padding: EdgeInsets.only(left: 0),
```
Comment (vladms) [inline marty_flutter/lib/screens/faq/faq.dart]: @MuBirhan Can't you  use EdgeInsets.zero ?

File: marty_flutter/lib/screens/faq/rotate_animation.dart [hunk 8]
```diff
@@ -4,7 +4,7 @@ class RotateAnimation extends StatefulWidget {
   final Widget child;
   final bool expand;
 
-  RotateAnimation({this.expand = false, this.child});
+  RotateAnimation(this.child, this.expand);
```
Comment (vladms) [inline marty_flutter/lib/screens/faq/rotate_animation.dart]: @MuBirhan¯  Why is this  called  RotateAnimation if it is a StatefullWidget?

**Comment by vladms** [comment]
@MuBirhan  Rename the faq.dart file to faq_screen.dart

Create the view model for faq screen which handles the data for faq screen

I see some overflow when the menu is fully exapnded. Please check

Increase space between faq title and arrow

**Pull request merged**